strategies/changed/src/main/java/org/arquillian/smart/testing/vcs/git/AllChangedFilesDetector.java

```diff
@@ -1,39 +1,50 @@
 package org.arquillian.smart.testing.vcs.git;
 
 import java.io.File;
-import java.util.HashSet;
+import java.util.Collection;
+import java.util.List;
 import java.util.Set;
+import org.eclipse.jgit.diff.DiffEntry;
 
 /**
- * The intention of this class is not to provide a final strategy to final user but to be used by other strategies that requires all changes of Git commits.
+ * The intention of this class is not to provide a final strategy to final user but to be used by other strategies that
+ * requires all changes of Git commits.
  */
-public class AllChangedFilesDetector {
+public class AllChangedFilesDetector extends GitChangesDetector {
 
-    private NewFilesDetector newFilesDetector;
-    private ChangedFilesDetector changedFilesDetector;
-
-    public AllChangedFilesDetector(File currentDir, String previous, String head, String ... globPatterns) {
-        newFilesDetector = new NewFilesDetector(currentDir, previous, head, globPatterns);
-        changedFilesDetector = new ChangedFilesDetector(currentDir, previous, head, globPatterns);
+    public AllChangedFilesDetector(File currentDir, String previous, String head, String... globPatterns) {
+        super(currentDir, previous, head, globPatterns);
     }
 
-    public Set<String> getTests() {
-        final Set<String> tests = new HashSet<>();
+    public Collection<String> getTests() {
+        final Collection<String> tests = super.getTests();
+
+        final Set<String> newFiles = this.gitChangeResolver.newChanges();
```

Comment:
I agree that it hardly makes sense to use `this` here, when we are actually making no reference to the current class. Thanks for pointing out @lordofthejars :)